libs/core/src/lib/utils/directives/carousel/carousel.directive.ts

```diff
@@ -0,0 +1,252 @@
+import {
+    AfterContentInit,
+    ChangeDetectorRef,
+    ContentChildren,
+    Directive,
+    ElementRef,
+    EventEmitter,
+    Input,
+    Output,
+    QueryList
+} from '@angular/core';
+import {
+    AnimationPlayer,
+} from '@angular/animations';
+import { CarouselItemDirective } from './carousel-item.directive';
+import * as Hammer from 'hammerjs';
+import { HammerConfig } from './carousel.module';
+import { Subject } from 'rxjs';
+
+
+export interface CarouselConfig {
+    vertical?: boolean;
+    elementsAtOnce?: number;
+    panSupport?: boolean
+    infinite?: boolean;
+    transition?: string;
+}
+
+export const Default_Transition_Duration: string = '150ms';
+
+
+@Directive({
+    selector: '[fdCarousel]',
+    host: {
+        class: 'fd-carousel'
+    }
+})
+export class CarouselDirective implements AfterContentInit {
+
+    /** Configuration for carousel */
+    @Input()
+    config: CarouselConfig;
+
+    /** Initial active item of carousel, position first + offset */
+    @Input()
+    active: CarouselItemDirective;
+
+    /** Whether support for mouse down, or tap should be enabled */
+    @Input()
+    panSupport: boolean = true;
+
+    /** Event thrown, when active element is changed */
+    @Output()
+    readonly activeChange: EventEmitter<CarouselItemDirective> = new EventEmitter<CarouselItemDirective>();
+
+    /** Event thrown, when element is started/stopped to be dragged */
+    @Output()
+    readonly dragged: EventEmitter<boolean> = new EventEmitter<boolean>();
```

Comment:
So I think the boolean value is for drag start vs. drag stop?  That is a bit unclear, I think it would make more sense to have 2 separate events.  Or at least more clearly document this

Reply:
I changed it to `dragStateChange` also documentation is corrected